File: src/main/java/com/example/chatService/dto/UserDto.java
package com.example.chatService.dto;

import com.example.chatService.entity.Role;
import com.example.chatService.entity.User;
import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@AllArgsConstructor
public class UserDto {
    private String id;
    private String username;
    private String email;
    private Role role;
    private boolean online;
    private boolean nicknameCompleted;

    public UserDto(String id, String username, String email, Role role, boolean nicknameCompleted) {
        this.id = id;
        this.username = username;
        this.email = email;
        this.role = role;
        this.nicknameCompleted = nicknameCompleted;
    }

    public static UserDto from(User user) {
        return new UserDto(
                user.getId(),
                user.getUsername(),
                user.getEmail(),
                user.getRole(),
                user.isNicknameCompleted()
        );
    }
}

File: src/main/java/com/example/chatService/dto/UserDto.kt
package com.example.chatService.dto

import com.example.chatService.entity.Role
import com.example.chatService.entity.User

data class UserDto(
    val id: String?,
    val username: String?,
    val email: String?,
    val role: Role?,
    val online: Boolean = false,
    val nicknameCompleted: Boolean = false
) {
    companion object {
        fun from(user: User): UserDto {
            return UserDto(
                id = user.id,
                username = user.username,
                email = user.email,
                role = user.role,
                nicknameCompleted = user.nicknameCompleted
            )
        }
    }
}
